apps/content-gateway-api/src/app/maintenance/jobs/index-handling/IndexCreationJob.ts

```diff
@@ -0,0 +1,146 @@
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import { MaintenanceJob, MaintenanceJobError } from "../..";
+import * as t from "io-ts";
+import { Db } from "mongodb";
+// @ts-expect-error There are no type definitions for this package :'(
+import * as DigestFetch from "digest-fetch";
+
+/**
+ * Queries MongoDB for index recommendations and then
+ * enacts those recommendations
+ */
+export const createIndexCreationJob = (
+    atlasApiInfo: AtlasApiInfo,
+    db: Db
+): MaintenanceJob => {
+    const runIndexCreation = (): TE.TaskEither<MaintenanceJobError, void> => {
+        return pipe(
+            _queryIndexSuggestions(atlasApiInfo),
+            TE.chain(_addIndexes(db))
+        );
+    };
+
+    return {
+        run: runIndexCreation,
+        id: "CreateIndexes",
+        schedule: {
+            hours: 1,
+        },
+    };
+};
+
+export interface AtlasApiInfo {
+    publicKey: string;
+    privateKey: string;
+    projectId: string;
+    processId: string;
+}
+
+/**
+ * INTERNAL OBJECTS, DO NOT USE:
+ * they are exported for testing purposes only
+ */
+
+/**io-ts codec for verifying external input*/
+const IndexSuggestionCodec = t.strict({
+    shapes: t.unknown, // We aren't using this so idc what it is
+    suggestedIndexes: t.array(
+        t.strict({
+            id: t.string,
+            impact: t.array(t.string),
+            index: t.array(t.record(t.string, t.number)),
+            namespace: t.string,
+            weight: t.number,
+        })
+    ),
+});
+
+export type IndexSuggestions = t.TypeOf<typeof IndexSuggestionCodec>;
+
+// TODO: remove hard coded process Id with automatic discovery of processes
+const makeSuggestedIndexUrl = (atlasApiInfo: AtlasApiInfo) =>
+    `https://cloud.mongodb.com/api/atlas/v1.0/groups/${atlasApiInfo.projectId}/processes/${atlasApiInfo.processId}/performanceAdvisor/suggestedIndexes`;
+
+/**
+ * Retrieves and verifies the request according to these specs
+ * https://docs.atlas.mongodb.com/reference/api/pa-suggested-indexes-get-all/
+ */
+export function _queryIndexSuggestions(
+    atlasApiInfo: AtlasApiInfo
+): TE.TaskEither<MaintenanceJobError, IndexSuggestions> {
+    const client = new DigestFetch(
+        atlasApiInfo.publicKey,
+        atlasApiInfo.privateKey,
+        {}
+    );
+    const url = makeSuggestedIndexUrl(atlasApiInfo);
+
+    // Query the endpoint for index suggestions
+    const unsafeFetch = (): Promise<Response> => client.fetch(url, {});
+    const fetch = (): TE.TaskEither<MaintenanceJobError, Response> =>
+        TE.tryCatch(
+            unsafeFetch,
+            (e) => new MaintenanceJobError(new Error(String(e)))
+        );
+
+    const parseResponse = (
+        res: Response
+    ): TE.TaskEither<MaintenanceJobError, IndexSuggestions> => {
+        return TE.tryCatch(
+            async () => {
+                const resJson = await res.json();
+                if (E.isLeft(IndexSuggestionCodec.decode(resJson)))
```

Comment:
You don't have to check if this is a `left`, you can just do a `TE.mapLeft` to convert the decoder error to a `MaintenanceJobError`.